sovryn-token-bridge/federator/src/lib/TransactionSender.js

```diff
@@ -50,9 +53,52 @@ module.exports = class TransactionSender {
     async createRawTransaction(from, to, data, value) {
         const nonce = await this.getNonce(from);
         const chainId =  this.chainId || await this.client.eth.net.getId();
-        const gasPrice = await this.getGasPrice(chainId);
-        let rawTx = {
-            gasPrice: this.numberToHexString(gasPrice),
+
+        console.log("chainId: " +chainId);
+
+        if (chainId === ETHERSCAN_CHAIN_ID ) {
+            const rawTxETH = await createETHRawTransaction(from, to, data, value);
+            console.log("rawTxETH: " + rawTxETH);
+            return rawTxETH;
+        }
+        else {
+            const gasPrice = await this.getGasPrice(chainId);
+            let rawTx = {
+                gasPrice: this.numberToHexString(gasPrice),
+                value: this.numberToHexString(value),
+                to: to,
+                data: data,
+                from: from,
+                nonce: this.numberToHexString(nonce),
+                r: 0,
+                s: 0
+            }
+            rawTx.gas = this.numberToHexString(await this.getGasLimit(rawTx));
+            return rawTx;
+        }
+    }
+    
+    async createETHRawTransaction(from, to, data, value) {
+        const nonce = await this.getNonce(from);
+        const gwei = 1000000000;
+        const priorityFee = 1 ;
+        const sleepOnGas = 10 * 1000 ; // 10 Seconds
+        const maxSleepOnGas = 12;
+        let sleepOnGasCounter = 0;
+        let rawTx;
+        console.log("raw tx is:" + rawTx);
+        while (currentEthGasBasePrice > currentEthGasPriceAvg) {
+            await utils.sleep(sleepOnGas, { logger: this.logger });
+            sleepOnGasCounter++;
+            if (sleepOnGasCounter > maxSleepOnGas) {
+                throw new CustomError(`High Base Gas: Transaction wasn't sent`);
+            }    
+        }        
+            
+            rawTx = {
+            //gasPrice: this.numberToHexString(gasPrice),
+            maxFeePerGas: this.numberToHexString((currentEthGasBasePrice + priorityFee) * gwei),
```

Comment:
Does etherscan API make it necessary to multiple by `gwei` (**Edit**: I think it does, or at least that's how GasPriceEstimator was coded in the past, but probably no harm double-checking it)? Have to be careful to not accidentally send 10000000000 times too much gas :D

Reply:
Yes, Etherscan receive wei units